web/frontend/src/components/navigation/components/scene.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang='ts'>
+
+import * as THREE from 'three';
+import { T, useThrelte, useRender, type CurrentWritable } from '@threlte/core';
+import { type Map } from 'maplibre-gl';
+import { obstacles, zoomLevels } from '../stores';
+import { createCameraTransform } from '../utils';
+import type { Mat4, Obstacle } from '../types';
+
+export let map: Map;
+export let viewProjectionMatrix: CurrentWritable<Float32Array | Mat4>;
+
+const { renderer, scene, camera } = useThrelte();
+
+renderer!.autoClear = false;
```

Comment:
@ethanlook @mcous  I'd usually agree, but this case is a quirk of Threlte's types, and one that I might try to advocate changing. It's because `renderer` is technically undefined until the `<Canvas>` mounts. Buuuuuut the `useThrelte()` hook that provides `renderer` is only available in child components of `<Canvas>`, which will not mount any of it's children until it itself is mounted. Because of this I think it's more desirable to error here since renderer should actually never be undefined.